src/LibVLCSharp/Shared/Core/Core.cs

```diff
@@ -72,7 +72,16 @@ static List<(string libvlccore, string libvlc)> ComputeLibVLCSearchPaths()
                 arch = PlatformHelper.IsX64BitProcess ? ArchitectureNames.Win64 : ArchitectureNames.Win86;
             }
 
-            var libvlcDirPath1 = Path.Combine(Path.GetDirectoryName(typeof(LibVLC).Assembly.Location),
+            var libvlcAssemblyLocation = typeof(LibVLC).Assembly.Location;
+
+            if (string.IsNullOrEmpty(libvlcAssemblyLocation)) /* .NET 5 and later */
+            {
+#if !NET40
```

Comment:
Why not around the if ?
Location should not be an empty string in NET40, but what if it actually does ? Shouldn't we check the returned values ?
For example, Path.GetDirectoryName may return null or an empty string : https://docs.microsoft.com/en-us/dotnet/api/system.io.path.getdirectoryname?view=net-5.0#System_IO_Path_GetDirectoryName_System_String_

Reply:
> Why not around the if ?

Doesn't really make a difference.

> Location should not be an empty string in NET40, but what if it actually does ? Shouldn't we check the returned values ?

We could, but while `Assembly.Location` _could_ be an empty string on an edge case
>  If the assembly is loaded from a byte array, such as when using Load(Byte[]), the value returned is an empty string ("")

`AppContext.BaseDirectory` being null or empty would mean the dotnet runtime can't function, seeing how pervasive its used in there.

> For example, Path.GetDirectoryName may return null or an empty string

> Directory information for path, or null if path denotes a root directory or is null. Returns Empty if path does not contain directory information.

Checking for this seems a bit too defensive for me. I'd say we wait until (if?) this problem occurs before we start checking for it?

Reply:
🆗